salt/modules/file.py

```diff
@@ -2676,16 +2691,33 @@ def _add_content(linesep, lines=None, include_marker_start=True,
             block_found = True
         elif append_if_not_found:
             # Make sure we have a newline at the end of the file
-            if 0 != len(new_file):
+            if new_file:
                 if not new_file[-1].endswith(linesep):
                     new_file[-1] += linesep
             # add the markers and content at the end of file
             _add_content(linesep, lines=new_file)
             block_found = True
-        else:
-            raise CommandExecutionError(
-                'Cannot edit marked block. Markers were not found in file.'
-            )
+        elif insert_before_match or insert_after_match:
+            match_regex = insert_before_match or insert_after_match
+            if not isinstance(match_regex, six.string_types):
+                raise CommandExecutionError(
+                    'RegEx expected in match parameter.'
+                )
+            match_idx = [i for i, item in enumerate(orig_file) if re.search(match_regex, item)]
+            if match_idx:
+                match_idx = match_idx[0]
+                for line in _add_content(linesep):
+                    if insert_after_match:
+                        match_idx += 1
+                    new_file.insert(match_idx, line)
+                    if insert_before_match:
+                        match_idx += 1
+                block_found = True
+
+    if not block_found:
+        raise CommandExecutionError(
+            'Cannot edit marked block. Markers were not found in file.'
+        )
 
     if block_found:
```

Comment:
This `if` is redundant. `block_found` is definitely `True` if we're here.

Reply:
@DmitryKuzmenko you will hit this section if you don't have any of the `append...`, `prepend...`, or `insert...` parameters set.

Reply:
ah. got it.

Reply:
@DmitryKuzmenko please take a look when you get a chance and see if this is what you intended.